sosw/components/test/unit/test_sns.py

```diff
@@ -124,5 +125,15 @@ def test_create_subscription_invalid_params(self):
         self.assertEqual(str(exc.exception), "You must send valid topic ARN, Protocol and Endpoint to add a subscription")
 
 
+    def test_get_message_attribute_validate_output(self):
+        self.assertEqual(self.sns.get_message_attribute(10), {'DataType': 'Number', 'StringValue': '10'})
+        self.assertEqual(self.sns.get_message_attribute(10.99), {'DataType': 'Number', 'StringValue': '10.99'})
+        self.assertEqual(self.sns.get_message_attribute('Test'), {'DataType': 'String', 'StringValue': 'Test'})
+        self.assertEqual(
+            self.sns.get_message_attribute(['Test1', 'Test2', 'Test3']),
+            {'DataType': 'String.Array', 'StringValue': json.dumps(['Test1', 'Test2', 'Test3'])}
+        )
+
+
```

Comment:
Need at least one more test that the autocommit works in case of changed attributes.
And the integration test described above.